upup/models/cloudup/resources/addons/authentication.aws/k8s-1.12.yaml.template

```
@@ -131,8 +131,8 @@ spec:
         {{- if or (not .Authentication.Aws.BackendMode) (contains "MountedFile" .Authentication.Aws.BackendMode) }}
         - --config=/etc/aws-iam-authenticator/config.yaml
         {{- end }}
-        {{- if or .Authentication.Aws.ClusterID (not (contains "MountedFile" .Authentication.Aws.BackendMode))}}
-        - --cluster-id={{ or .Authentication.Aws.ClusterID .MasterPublicName }}
+        {{- if or .Authentication.Aws.ClusterID (not .Authentication.Aws.BackendMode) (not (contains "MountedFile" .Authentication.Aws.BackendMode))}}
```

I'm not sure i understand the logic change here.

The old logic would add `--cluster-id` if any of the following:
* ClusterID API field is set
* BackendMode does not contain MountedFile (including if BackendMode is not set)

This new logic will add `--cluster-id` if any of the following:
* ClusterID API field is set
* BackendMode API field is not set
* BackendMode API field does not contain MountedFile

Wouldn't we want to _not_ set --cluster-id if BackendMode is not set in order to preserve original behavior? 

```suggestion
        {{- if or .Authentication.Aws.ClusterID (and .Authentication.Aws.BackendMode (not (contains "MountedFile" .Authentication.Aws.BackendMode)))}}
```

The above would add `--cluster-id` if any of the following:
* ClusterID API field is set
* BackendMode API field is set and does not contain MountedFile

---

Yeah, you're right. In my local tests I saw the `--cluser-id` flag get replaced with the `--config` and didn't even notice the clusterID flag was just a few lines lower. I've tested all the different possibilities you mentioned here and with your suggested change everything is working as expected. Thanks for catching that.